api/WebApi.ts

```diff
@@ -90,17 +91,19 @@ export class WebApi {
         this.authHandler = authHandler;
         this.options = options || {};
 
-        // try get proxy setting from environment variable set by VSTS-Task-Lib if there is no proxy setting in the options
-        if (!this.options.proxy || !this.options.proxy.proxyUrl) {
-            if (global['_vsts_task_lib_proxy']) {
-                let proxyFromEnv: VsoBaseInterfaces.IProxyConfiguration = {
-                    proxyUrl: global['_vsts_task_lib_proxy_url'],
-                    proxyUsername: global['_vsts_task_lib_proxy_username'],
-                    proxyPassword: this._readTaskLibSecrets(global['_vsts_task_lib_proxy_password']),
-                    proxyBypassHosts: JSON.parse(global['_vsts_task_lib_proxy_bypass'] || "[]"),
-                };
-
-                this.options.proxy = proxyFromEnv;
+        if (!process.env.no_proxy || process.env.no_proxy.toLowerCase().split(',').indexOf(url.parse(this.serverUrl).host) !== -1) {
```

Comment:
@gforceg @nschonni Has this been tested to make sure it works as desired? I think this is backwards. It should be === instead of !==, right? We are trying to enter this block of code only if the server host is not in no_proxy.

```
        var no_proxy = 'www.google.com';
        var serverUrlHost = 'www.google.com';

        if (!no_proxy || no_proxy.toLowerCase().split(',').indexOf(serverUrlHost) === -1) {
            console.log('allowed setting proxy');
        }
        else {
            console.log('skipped setting proxy');
        }
```

Also probably want toLowerCase on serverUrl if url.parse doesn't do that for you.

I'm not opposed to extracting the logic to a function.

Reply:
Think you're right. Looking at https://github.com/request/request/blob/b12a6245d9acdb1e13c6486d427801e123fdafae/lib/getProxyFromURI.js there may be some additional logic needed too

Reply:
Thanks for the link Nick, appreciate it. IMO all the more reason to extract this to a method.

@gforceg Could you make the suggested changes? Thanks! Appreciate the contribution.

Reply:
@stephenmichaelf, will do, thanks!

Reply:
@gforceg Any updates on this work?

Reply:
@gforceg any updates? Are you still planning on submitting this PR?

Reply:
Sorry for the delay on this.
First I was on vacation then I was doing other work.

I wrapped the functionality into a function and then rebased.

Can you provide guidance on where I can stick a unit test?
I samples/common.ts mentions proxy but I'm still trying to understand the how the samples / tests work in this project.

Reply:
haha I just noticed the tests directory.
Added a unit test. Still needs to be e2e tested.
@nschonni, can you give this a try?